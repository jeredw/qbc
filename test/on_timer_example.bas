    ON TIMER(1) GOSUB TimeUpdate
    TIMER ON
    CLS
    PRINT "Time: "; TIME$
    StartTime = TIMER
    WHILE TimePast < 10
        TimePast = TIMER - StartTime
    WEND
    END

    TimeUpdate:
        LOCATE 1, 7: PRINT TIME$
        RETURN
