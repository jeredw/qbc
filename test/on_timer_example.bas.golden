(program 
  (statement 
    (on_event_gosub_statement ON TIMER 
      ( 
        (expr 
          (literal 1)) ) GOSUB 
      (target 
        (text_label 
          (untyped_id TimeUpdate))))) \n 
  (statement 
    (event_control_statement TIMER ON)) \n 
  (statement 
    (call_statement 
      (untyped_id CLS))) \n 
  (statement 
    (print_statement PRINT 
      (expr 
        (literal "Time: ")) ; 
      (expr 
        (variable_or_function_call TIME$)))) \n 
  (statement 
    (assignment_statement 
      (variable_or_function_call StartTime) = 
      (expr 
        (builtin_function TIMER)))) \n 
  (statement 
    (while_wend_statement WHILE 
      (expr 
        (expr 
          (variable_or_function_call TimePast)) < 
        (expr 
          (literal 10))) 
      (block \n 
        (statement 
          (assignment_statement 
            (variable_or_function_call TimePast) = 
            (expr 
              (expr 
                (builtin_function TIMER)) - 
              (expr 
                (variable_or_function_call StartTime))))) \n statement) WEND)) \n 
  (statement 
    (end_statement END)) \n statement \n 
  (label 
    (text_label 
      (untyped_id TimeUpdate)) :) statement \n 
  (statement 
    (call_statement 
      (untyped_id LOCATE) 
      (call_argument_list 
        (call_argument 
          (expr 
            (literal 1))) , 
        (call_argument 
          (expr 
            (literal 7)))))) : 
  (statement 
    (print_statement PRINT 
      (expr 
        (variable_or_function_call TIME$)))) \n 
  (statement 
    (return_statement RETURN)) \n)
